src/main/java/com/pirimid/cryptotrade/service/Impl/OrderServiceImpl.java

```diff
@@ -57,61 +108,139 @@ public Order getOrderById(UUID id) {
-            order.setSymbol(placeOrderResDTO.getSymbol());
-            order.setPrice(placeOrderResDTO.getPrice());
-            order.setFilledQuantity(placeOrderResDTO.getExecutedAmount());
-            order.setAccount(account);
-            orderRepository.save(order);
-            return  placeOrderResDTO;
+        if (!optAccount.isPresent()) {
+            return null;
+        }
+        Account account = optAccount.get();
+        OrderResDTO orderResDTO = exchangeUtil
+                .getObject(EXCHANGE.valueOf(account.getExchange().getName().toUpperCase()))
+                .createOrder(account.getApiKey(), account.getSecretKey(), account.getPassPhrase(), req);
+        if (orderResDTO == null) {
+            return null;
+        }
+        orderResDTO.setAccountId(optAccount.get().getAccountId());
+        return orderResDTO;
+    }
+
+    @Override
+    public OrderResDTO createOrder(OrderResDTO orderDto) {
+        Optional<Account> optAccount = accountRepository.findById(orderDto.getAccountId());
+        if (!optAccount.isPresent()) {
+            return null;
+        }
+        Optional<Order> order = orderRepository.findByOrderIdExchangeAndAccount_Exchange(orderDto.getExchangeOrderId(), optAccount.get().getExchange()); // accoount->exchange
+        if (order.isPresent()) {
+            orderDto.setOrderId(order.get().getOrderId());
+            return orderDto;
+        }
+        Order newOrder = orderResDtoToOrder(orderDto,optAccount.get());
+        newOrder = orderRepository.save(newOrder);
+        orderDto.setOrderId(newOrder.getOrderId());
+        orderDto.setAccountId(newOrder.getAccount().getAccountId());
+        return orderDto;
+    }
+
+    @Override
+    public OrderResDTO addTrade(TradeDto tradeDto,EXCHANGE exchange) {
+        Optional<Exchange> optExchange = exchangeRepository.findByName(exchange.getValue().toLowerCase());
+        if(!optExchange.isPresent()) return null;
+        Optional<Order> optOrder = orderRepository.findByOrderIdExchangeAndAccount_Exchange(tradeDto.getExchangeOrderId(),optExchange.get()); // add exchange name
+        if (!optOrder.isPresent()) return null;     // no such order exists
+
+        Trade trade = new Trade();
+        trade.setAmount(tradeDto.getFunds());
+        trade.setMarketPrice(tradeDto.getPrice());
+        trade.setQuantity(tradeDto.getSize());
+        trade.setTimestamp(tradeDto.getTime());
+        trade.setTradeIdExchange(tradeDto.getTradeId());
+
+        Order order = optOrder.get();
+        order.setCommission((order.getCommission() == null ? 0 : order.getCommission()) + tradeDto.getFee());
+        order.setOrderStatus(Status.PARTIALLY_FILLED);
+
+        if (order.getOrderType().equals(OrderType.MARKET))
+            order.setFilledQuantity((order.getFilledQuantity() == null ? 0 : order.getFilledQuantity()) + (order.getOrderQty() == null ? tradeDto.getFunds() : tradeDto.getSize()));
+         else //limit
```

Comment:
Can you write if condition for 'LIMIT' as well as in worst case scenario if value is nether from Market, Limit then also limit order will be placed

Reply:
okay got it